programs/aes/CMakeLists.txt

```diff
@@ -1,10 +1,12 @@
+find_package(Threads)
```

Comment:
Why don't we set `THREADS_PREFER_PTHREAD_FLAG` before this?

Reply:
Good point - the flag is relatively new, I think that is the main issue. Have set both this and to prefer PThreads, as it will cover more corner cases.